python/tests/test_types.py

```diff
@@ -129,6 +150,11 @@ def test_map_type():
     assert isinstance(type_var.value.type, UUIDType)
     assert type_var.value.field_id == 2
     assert str(type_var) == str(eval(repr(type_var)))
+    assert type_var == eval(repr(type_var))
+    assert type_var != MapType(
+        NestedField(True, 1, "optional_field", LongType()),
+        NestedField(False, 2, "required_field", UUIDType()),
+    )
```

Comment:
should there be another test case to test inequality of the Value NestedField as well?

Reply:
Agreed.

Reply:
Yeah, added it. Thanks for the comment.